src/utils/time.ts

```diff
@@ -169,6 +177,43 @@ export const formatHourAndMinute = (utcSeconds: number) => {
   return format(utcSeconds * ONE_SECOND_IN_MILLISECONDS, "HH:mm");
 };
 
+// TODO: do we need it?
+export const formatHour = (utcSeconds: number) => {
+  const date = new Date(utcSeconds * ONE_SECOND_IN_MILLISECONDS);
+  const hh = String(date.getHours()).padStart(2, "0");
+  return hh;
+};
+
+// TODO: do we need it?
+export const getMinutes = (utcSeconds: number) => {
+  const date = new Date(utcSeconds * ONE_SECOND_IN_MILLISECONDS);
+  const hh = date.getHours();
+  const mm = date.getMinutes();
+  const minutes = hh * 60 + mm;
+  return minutes;
+};
+
+export const getSecondsFromStartOfDay = (utcSeconds: number) => {
+  const utcMilliseconds = utcSeconds * ONE_SECOND_IN_MILLISECONDS;
+
+  return (
+    getHours(utcMilliseconds) * ONE_HOUR_IN_SECONDS +
+    getLibMinutes(utcMilliseconds) * ONE_MINUTE_IN_SECONDS +
+    getSeconds(utcMilliseconds)
+  );
+};
```

Comment:
What is this function intended to do? Is it getting the number of seconds between the start of the day and the provided `utcSeconds` timestamp?

If so, can we just do something like this?

- https://date-fns.org/v2.20.1/docs/startOfToday
- https://date-fns.org/v2.20.1/docs/differenceInSeconds

```suggestion
export const getSecondsFromStartOfDay = (utcSeconds: number) => {
  const utcMilliseconds = utcSeconds * ONE_SECOND_IN_MILLISECONDS;

  return differenceInSeconds(startOfToday(), utcMilliseconds);
};
```

Reply:
yeah, the way you suggest to rewrite it looks really better. Thanks.
Although it's not necessarily need to be `startOfToday()` but the result should be a positive number of seconds from the start of any day (so I used `startOfDay()`).

see the changes here: https://github.com/sparkletown/sparkle/pull/1302/commits/4620ee606b4d082901e989fc6eac27f0e5b9a265

Reply:
Is the order of the parameters correct in 4620ee6? (not sure if it will always be a positive number, or if it can be negative)

You have `differenceInSeconds(utcMilliseconds, startOfDay(utcMilliseconds));`

Mine was `differenceInSeconds(startOfDay(utcMilliseconds), utcMilliseconds)`

Reply:
if we look at the example in the documentation https://date-fns.org/v1.29.0/docs/differenceInSeconds#examples

```js
// How many seconds are between
// 2 July 2014 12:30:07.999 and 2 July 2014 12:30:20.000?
var result = differenceInSeconds(
  new Date(2014, 6, 2, 12, 30, 20, 0),
  new Date(2014, 6, 2, 12, 30, 7, 999)
)
//=> 12
```

we can see that the first date is minuend (which should be bigger - utc seconds), and the second is subtrahend (smaller - start of the day).

Reply:
Yeah, makes sense.

The docs confirm as well:
`node_modules/date-fns/differenceInSeconds/index.js:25`
```
* @param {Date|Number} dateLeft - the later date
* @param {Date|Number} dateRight - the earlier date
```